Subject: Linux-RT 2.6.24-rc8-rt1
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
Subject: [PATCH] just rename call_rcu_bh instead of making it a macro

Seems that I found a box that has a config that passes call_rcu_bh as a
function pointer (see net/sctp/sm_make_chunk.c), so declaring the
call_rcu_bh has a macro function isn't good enough.

This patch makes it just another name of call_rcu for rcupreempt.

Signed-off-by: Steven Rostedt <rostedt@goodmis.org>

---
 include/linux/rcupdate.h   |    4 ++--
 include/linux/rcupreempt.h |    7 ++++++-
 2 files changed, 8 insertions(+), 3 deletions(-)

Index: linux-2.6.24-rc8-rt1/include/linux/rcupreempt.h
===================================================================
--- linux-2.6.24-rc8-rt1.orig/include/linux/rcupreempt.h	2008-01-16 22:28:23.000000000 -0500
+++ linux-2.6.24-rc8-rt1/include/linux/rcupreempt.h	2008-01-16 22:29:17.000000000 -0500
@@ -42,7 +42,12 @@
 #include <linux/cpumask.h>
 #include <linux/seqlock.h>
 
-#define call_rcu_bh(head, rcu) call_rcu(head, rcu)
+/*
+ * Someone might want to pass call_rcu_bh as a function pointer.
+ * So this needs to just be a rename and not a macro function.
+ *  (no parentheses)
+ */
+#define call_rcu_bh	 	call_rcu_preempt
 #define rcu_bh_qsctr_inc(cpu)	do { } while (0)
 #define __rcu_read_lock_bh()	{ rcu_read_lock(); local_bh_disable(); }
 #define __rcu_read_unlock_bh()	{ local_bh_enable(); rcu_read_unlock(); }
Index: linux-2.6.24-rc8-rt1/include/linux/rcupdate.h
===================================================================
--- linux-2.6.24-rc8-rt1.orig/include/linux/rcupdate.h	2008-01-16 22:28:21.000000000 -0500
+++ linux-2.6.24-rc8-rt1/include/linux/rcupdate.h	2008-01-16 22:29:17.000000000 -0500
@@ -221,9 +221,9 @@ extern struct lockdep_map rcu_lock_map;
  * and may be nested.
  */
 #ifdef CONFIG_CLASSIC_RCU
-#define call_rcu(head, func) call_rcu_classic(head, func)
+#define call_rcu call_rcu_classic
 #else /* #ifdef CONFIG_CLASSIC_RCU */
-#define call_rcu(head, func) call_rcu_preempt(head, func)
+#define call_rcu call_rcu_preempt
 #endif /* #else #ifdef CONFIG_CLASSIC_RCU */
 
 /**
